business/istio_config.go

```diff
@@ -473,3 +479,96 @@ func getUpdateDeletePermissions(k8s kubernetes.IstioClientInterface, namespace,
 	}
 	return canUpdate || canPatch, canDelete
 }
+
+func (in *IstioConfigService) MeshWidemTLSStatus() (string, error) {
+	mpp, mpErr := in.hasMeshPolicyEnabled()
+	if mpErr != nil {
+		return "", mpErr
+	}
+
+	drp, drErr := in.hasDestinationRuleEnabled()
+	if drErr != nil {
+		return "", drErr
+	}
+
+	if drp && mpp {
+		return MeshmTLSEnabled, nil
+	} else if drp || mpp {
+		return MeshmTLSPartiallyEnabled, nil
+	}
+
+	return MeshmTLSNotEnabled, nil
+}
+
+func (in *IstioConfigService) hasMeshPolicyEnabled() (bool, error) {
+	mps, err := in.k8s.GetMeshPolicies()
+	if err != nil {
+		return false, err
+	}
+
+	mtlsEnabled := false
+
+	for _, mp := range mps {
+
+		// It is mandatory to have default as a name
+		if meshMeta := mp.GetObjectMeta(); meshMeta.Name != "default" {
+			continue
+		}
+
+		// It is no globally enabled when has targets
+		targets, targetPresent := mp.GetSpec()["targets"]
+		specificTarget := targetPresent && len(targets.([]interface{})) > 0
+		if specificTarget {
+			continue
+		}
+
+		// It is globally enabled when a peer has mtls enabled
+		peers, peersPresent := mp.GetSpec()["peers"]
+		if !peersPresent {
+			continue
+		}
+
+		for _, peer := range peers.([]interface{}) {
+			peerMap := peer.(map[string]interface{})
+			if _, present := peerMap["mtls"]; present {
+				mtlsEnabled = true
+				break
+			}
+		}
+	}
+
+	return mtlsEnabled, nil
+}
+
+func (in *IstioConfigService) hasDestinationRuleEnabled() (bool, error) {
+	drs, err := in.k8s.GetAllDestinationRules()
+	if err != nil {
+		return false, err
+	}
+
+	mtlsEnabled := false
+
+	for _, dr := range drs {
+		// Host my be *.local
```

Comment:
I didn't see this before.
Can this wildcard be different ? I mean someting like *.svc.local or other combination ?
Perhaps if this is documented in a istio.io doc page I'd add a link in the comment - no blocker - but that would help to review & maintain in next iterations.